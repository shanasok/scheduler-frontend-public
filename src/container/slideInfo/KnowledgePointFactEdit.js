import React from 'react';
import {bindActionCreators, compose} from "redux";
import {withRouter} from "react-router-dom";
import {connect} from "react-redux";
import {setCurrentLessonContent} from "../../redux/actions/lessonContentAction";
import TeacherSavePageContentIcon from "../teacher/TeacherSavePageContentIcon";


class KnowledgePointFactEdit extends React.Component {

    constructor(props) {
        super(props);

        this.state = {
            newVal: (this.props.content? this.props.content.fact : ""),
        };

        this.updateLocalState = this.updateLocalState.bind(this);
    }

    updateLocalState(event) {
        this.setState({newVal: event.target.value});
    }

    render() {

        const fact = (this.props.content)? this.props.content.fact : "";
        const elementIdForInput = "factEditBox"+this.props.contentId;

        return <div className="knowledgePoint">
            <TeacherSavePageContentIcon
                contentId={this.props.contentId}
                newValues={{
                    'fact':this.state.newVal
                }}
            />
            <input defaultValue={fact}
                   className="factEditBox"
                   id={elementIdForInput}
                   onChange={(e) => this.updateLocalState(e)}
            />

        </div>
    }
}

const mapStateToProps = state => ({
    currentLessonContent: state.currentLessonContent,
});

function mapDispatchToProps(dispatch) {
    return {
        actions: bindActionCreators({
            setCurrentLessonContent,
        }, dispatch)
    }
}

export default compose(
    withRouter,
    connect(mapStateToProps, mapDispatchToProps)
)(KnowledgePointFactEdit);
