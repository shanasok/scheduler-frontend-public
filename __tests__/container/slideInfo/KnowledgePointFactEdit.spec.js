import React from 'react';
import KnowledgePointFactEdit from '../../../src/container/slideInfo/KnowledgePointFactEdit';
import renderer from 'react-test-renderer';
import configureMockStore from "redux-mock-store";
import {configure, mount} from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import {Provider as ReduxProvider} from "react-redux";
import {BrowserRouter} from "react-router-dom";
import {MemoryRouter} from "react-router";

const lesson = {
    id: 1,
    name: "test lesson name",
    lessonNumber: 1,
    pages: []
}

const mockStore = configureMockStore()
const mockState = {
    lesson: lesson
}
let store = mockStore(mockState)
configure({adapter: new Adapter()});

it('renders correctly when fact is given', () => {

    const content = {
        contentType: "FACT",
        fact: "Test Fact",
        id: "1",
    }

    const tree = renderer
        .create(<BrowserRouter><ReduxProvider store={store}><KnowledgePointFactEdit content={content}/></ReduxProvider></BrowserRouter>)
        .toJSON();
    expect(tree).toMatchSnapshot();
    expect(tree.children[1].props['defaultValue']).toContain(content.fact);
});

it('renders correctly when fact is not given', () => {

    const tree = renderer
        .create(<BrowserRouter><ReduxProvider store={store}><KnowledgePointFactEdit /></ReduxProvider></BrowserRouter>)
        .toJSON();
    expect(tree).toMatchSnapshot();
    expect(tree.children[1].props['defaultValue']).toContain("");
});


it("should call setState with new fact val", () => {
    const content = {
        'newVal':'testVal',
    }

    const mockEventFact = {
        target: {
            value: "newTestVal"
        }
    };

    const expected = {
        'newVal':'newTestVal',
    }

    let wrapper = mount(
        <MemoryRouter initialEntries={[ '/' ]}>
            <ReduxProvider store={store}>
                <KnowledgePointFactEdit content={content}/>
            </ReduxProvider>
        </MemoryRouter>
    );

    wrapper.find('.factEditBox').simulate('change',mockEventFact);
    wrapper.instance().updateLocalState = jest.fn();
    wrapper.instance().updateLocalState(mockEventFact);
    expect(wrapper.instance().updateLocalState).toHaveBeenCalledWith(mockEventFact);

    expect(wrapper.childAt(0).childAt(0).childAt(0).childAt(0).childAt(0)
        .instance().state).toEqual(expected);
});