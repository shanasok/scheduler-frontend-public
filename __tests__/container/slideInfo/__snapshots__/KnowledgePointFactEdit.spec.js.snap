// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when fact is given 1`] = `
<div
  className="knowledgePoint"
>
  <i
    className="fas fa-save mouseOverBecomeArrow"
    onClick={[Function]}
  />
  <input
    className="factEditBox"
    defaultValue="Test Fact"
    id="factEditBoxundefined"
    onChange={[Function]}
  />
</div>
`;

exports[`renders correctly when fact is not given 1`] = `
<div
  className="knowledgePoint"
>
  <i
    className="fas fa-save mouseOverBecomeArrow"
    onClick={[Function]}
  />
  <input
    className="factEditBox"
    defaultValue=""
    id="factEditBoxundefined"
    onChange={[Function]}
  />
</div>
`;
